src/hardware/vga_other.cpp

```diff
@@ -359,33 +359,26 @@ static void update_cga16_color(void) {
 	// PCjr CGA palette table
 	const uint8_t PCJRpal[] = {vga.attr.palette[0], vga.attr.palette[1],
 	                   vga.attr.palette[2], vga.attr.palette[3]};
+
+	const auto machine_palette = (machine == MCH_PCJR) ? PCJRpal : CGApal;
+
 	for (Bit8u x = 0; x < 4; x++) { // Position of pixel in question
-		bool even = (x & 1) == 0;
+		const bool even = !(x & 1);
 		for (Bit8u bits=0; bits<(even ? 0x10 : 0x40); ++bits) {
 			double Y=0, I=0, Q=0;
 			for (Bit8u p=0; p<4; p++) {  // Position within color carrier cycle
 				// generate pixel pattern.
-				Bit8u rgbi;
+				uint8_t rgbi = 0;
 				if (bpp1) {
 					rgbi = ((bits >> (3 - p)) & (even ? 1 : 2)) != 0
 					               ? overscan
 					               : 0;
 				} else {
-					if (machine == MCH_PCJR) {
-						if (even) {
-							rgbi = PCJRpal[(bits >> (2 - (p & 2))) & 3];
-						} else {
-							rgbi = PCJRpal[(bits >>
-							                (4 - ((p + 1) & 6))) &
-							               3];
-						}
-					} else {
-						if (even) {
-							rgbi = CGApal[(bits >> (2 - (p & 2))) & 3];
-						} else {
-							rgbi = CGApal[(bits >> (4 - ((p + 1) & 6))) & 3];
-						}
-					}
+					const size_t index =
```

Comment:
Use different variable name than `index` - GitHub colorizes is as if it was POSIX function `index`. Even `i` or `ii` is fine here - scope of the variable is super short.